src/backlight/portfolio/portfolio.py

```diff
@@ -72,24 +72,73 @@ def construct_portfolio(
         Portfolio
     """
 
+    assert len(principal) == len(trades)
+    assert len(lot_size) == len(trades)
+
     symbols2mkt = {m.symbol: m for m in mkt}
     symbols = [t.symbol for t in trades]
     assert set(symbols) == set(symbols2mkt.keys())
 
     # Transform trades following the lot_size
+    mult_trades = []
     for (trade, lot) in zip(trades, lot_size):
-        trade["amount"] *= lot
+        mult_trade = trade.copy()
+        mult_trade["amount"] *= lot
+        mult_trades.append(mult_trade)
 
     # Construct positions and return Portfolio
     positions = Parallel(n_jobs=-1, max_nbytes=None)(
         [
-            delayed(calc_positions)(trade, market, principal=principal_per_asset)
-            for (trade, market, principal_per_asset) in zip(trades, mkt, principal)
+            delayed(calc_positions)(
+                trade, symbols2mkt[trade.symbol], principal=principal_per_asset
+            )
+            for (trade, principal_per_asset) in zip(mult_trades, principal)
         ]
     )
+
+    symbols = [p.symbol for p in positions]
+    if len(set(symbols)) != len(symbols):
+        positions = _fusion_positions(positions)
+
     return Portfolio(positions)
 
 
+def _fusion_positions(positions: List[Positions]) -> List[Positions]:
+    """
+    Take a list of Positions and sum those with the same symbols
+    args :
+        - positions : a list of Positions to fusion
+    """
+
+    unique_positions = []
+    symbols = [p.symbol for p in positions]
+    columns = positions[0].columns
+
+    for symbol in sorted(set(symbols)):
+        positions_of_symbol = [p for p in positions if p.symbol == symbol]
+        cross_index = positions_of_symbol[0].index.copy()
+        for p in positions_of_symbol:
+            cross_index = pd.DatetimeIndex.union(cross_index, p.index)
+
+        df = pd.DataFrame(
+            np.array(
+                [
+                    [
+                        (p.loc[idx, :].values if idx in p.index else [0] * len(columns))
+                        for idx in cross_index
+                    ]
+                    for p in positions_of_symbol
+                ]
+            ).sum(axis=0),
+            index=cross_index,
+            columns=columns,
+        )
+        position = position_from_dataframe(df, symbol)
```

Comment:
I want to align naming conventions, so please use `from_dataframe`. If you  write `import backlight.positions.positions`, you can write `backlight.positions.positions.from_datafram`.